agent/src/softwarecontaineragent.cpp

```diff
@@ -79,15 +79,15 @@ SoftwareContainerAgent::SoftwareContainerAgent(
         defaultServiceManifestDir = config.getStringValue(Config::SC_GROUP,
                                                           Config::DEFAULT_SERVICE_MANIFEST_DIR_KEY);
     } catch (ConfigError &error) {
-        throw ReturnCode::FAILURE;
+        throw;
     }
 
     try {
         m_filteredConfigStore = std::make_shared<FilteredConfigStore>(serviceManifestDir);
         m_defaultConfigStore = std::make_shared<DefaultConfigStore>(defaultServiceManifestDir);
     } catch (ReturnCode err) {
         log_error() << "Failed to initialize ConfigStore";
-        throw ReturnCode::FAILURE;
+        throw SoftwareContainerAgentError();
```

Comment:
It would make sense to use the message that we log as message to the exception also, in all places in this file.